### PR TITLE
[MAT-1042][MAT-1045] Parsing Multiple Population Sets, Observations, and Stratifications

### PR DESCRIPTION
**Pull request by jkotanchik-SB**
- Added parsing logic for multiple population sets, observations, and stratifications
- Unit test for Observation. Tests for multiple population sets and stratifications in progress.
- Commented out checking of source data criteria in extract measure unit test with note to uncomment once TS models have been implemented.
- Corrected issues with value set loader test.

**Submitter:**
- [x] This pull request describes why these changes were made.
- [x] Internal ticket for this PR: [MAT-1042](https://jira.cms.gov/browse/MAT-1042), [MAT-1045](https://jira.cms.gov/browse/MAT-1045)
- [ ] Internal ticket links to this PR
- [ ] Code diff has been done and been reviewed
- [ ] Tests are included and test edge cases
- [x] Tests have been run locally and pass

**Reviewer 1:**

Name:
- [ ] Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
- [ ] The tests appropriately test the new code, including edge cases
- [ ] You have tried to break the code

**Reviewer 2:**

Name:
- [ ] Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
- [ ] The tests appropriately test the new code, including edge cases
- [ ] You have tried to break the code

### CODE REVIEW

File: test/unit/measure-loader/bundle_loader_test.rb [hunk 1]
```diff
@@ -26,8 +26,11 @@ def test_extract_measure
       # value_set_ids doesn't come with the embeds_many :value_sets relation.
       # assert_equal 46, measure.value_set_ids.count, 'Mismatching number of value set Ids.'
       assert_equal 5, measure.cql_libraries.size, 'Mismatching number of cql libraries.'
-      assert_equal 10, measure.source_data_criteria.length, 'Mismatching number of source_data_criteria.'
-      assert_equal [CQM::DataElement], measure.source_data_criteria.map(&:class).uniq, 'Mismatching source_data_criteria object type.'
+
+      # TODO: uncomment once we have TS models integrated in bonnie.
```
Comment (serhii-ilin) [inline test/unit/measure-loader/bundle_loader_test.rb:30]: when can we try TS models?
